dts/bindings/adc/adc-controller.yaml

```diff
@@ -12,3 +13,102 @@ properties:
     "#io-channel-cells":
       type: int
       required: true
+
+    "#address-cells":
+      required: false
+      const: 1
+
+    "#size-cells":
+      required: false
+      const: 0
+
+child-binding:
+    description: |
+      Channel configuration. See the adc_channel_cfg structure in adc.h.
+
+      All nodes using this binding must be named "channel", otherwise their
+      data will not be accessible by the ADC_CHANNEL_DT_NODE macro.
+
+    properties:
+      reg:
+        type: array
+        required: true
+        description: Channel identifier.
+
+      gain:
+        type: string
+        required: true
+        description: Gain selection.
+        enum:
+          - "ADC_GAIN_1_6" # x 1/6.
+          - "ADC_GAIN_1_5" # x 1/5.
+          - "ADC_GAIN_1_4" # x 1/4.
+          - "ADC_GAIN_1_3" # x 1/3.
+          - "ADC_GAIN_1_2" # x 1/2.
+          - "ADC_GAIN_2_3" # x 2/3.
+          - "ADC_GAIN_1"   # x 1.
+          - "ADC_GAIN_2"   # x 2.
+          - "ADC_GAIN_3"   # x 3.
+          - "ADC_GAIN_4"   # x 4.
+          - "ADC_GAIN_6"   # x 6.
+          - "ADC_GAIN_8"   # x 8.
+          - "ADC_GAIN_12"  # x 12.
+          - "ADC_GAIN_16"  # x 16.
+          - "ADC_GAIN_24"  # x 24.
+          - "ADC_GAIN_32"  # x 32.
+          - "ADC_GAIN_64"  # x 64.
+          - "ADC_GAIN_128" # x 128.
+
+      reference:
+        type: string
+        required: true
+        description: Reference selection.
+        enum:
+          - "ADC_REF_VDD_1"     # VDD.
+          - "ADC_REF_VDD_1_2"   # VDD/2.
+          - "ADC_REF_VDD_1_3"   # VDD/3.
+          - "ADC_REF_VDD_1_4"   # VDD/4.
+          - "ADC_REF_INTERNAL"  # Internal.
+          - "ADC_REF_EXTERNAL0" # External, input 0.
+          - "ADC_REF_EXTERNAL1" # External, input 1.
+
+      acquisition_time:
```

Comment:
Please use dashes when naming properties in DTS; this is the convention. In this case,  it should be `acquisition-time` instead.

Reply:
Fixed.